src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -373,8 +373,13 @@ public Blob findOrCreateByPath(String tenantId, String path) {
             throw new IllegalArgumentException("An empty path was provided!");
         }
 
-        return blobByPathCache.get(determinePathCacheKey(tenantId, path),
-                                   ignored -> fetchOrCreateByPath(tenantId, path));
+        String key = determinePathCacheKey(tenantId, path);
+        Blob blob = blobByPathCache.get(key);
+        if (blob == null) {
+            blobByPathCache.remove(key);
```

Comment:
why not
```
 Blob blob = fetchOrCreateByPath(tenantId, path);
blobByPathCache.put(key, blob);

Reply:
+-

Reply:
Unfortunately put won't sync a coherent cache